fontbe/src/fvar.rs

```diff
@@ -0,0 +1,72 @@
+//! Generates a [fvar](https://learn.microsoft.com/en-us/typography/opentype/spec/fvar) table.
+
+use std::collections::HashMap;
+
+use fontdrasil::orchestration::Work;
+use log::trace;
+use read_fonts::types::{Fixed, MajorMinor, Tag};
+use write_fonts::tables::fvar::{AxisInstanceArrays, Fvar, VariationAxisRecord};
+
+use crate::{
+    error::Error,
+    orchestration::{BeWork, Context},
+};
+
+struct FvarWork {}
+
+pub fn create_fvar_work() -> Box<BeWork> {
+    Box::new(FvarWork {})
+}
+
+impl Work<Context, Error> for FvarWork {
+    /// Generate [fvar](https://learn.microsoft.com/en-us/typography/opentype/spec/fvar)
+    fn exec(&self, context: &Context) -> Result<(), Error> {
+        let static_metadata = context.ir.get_init_static_metadata();
+        // Guard clause: don't produce fvar for a static font
+        if static_metadata.axes.is_empty() {
+            trace!("Skip fvar; this is not a variable font");
+            return Ok(());
+        }
+
+        let reverse_names: HashMap<_, _> = static_metadata
+            .names
+            .iter()
+            .map(|(key, name)| (name, key.name_id))
+            .collect();
+
+        let axes_and_instances = AxisInstanceArrays::new(
+            static_metadata
+                .axes
+                .iter()
+                .map(|ir_axis| {
+                    let axis_name_id: u16 = (*reverse_names.get(&ir_axis.name).unwrap()).into();
+                    let mut var = VariationAxisRecord {
+                        axis_tag: Tag::new(ir_axis.tag.as_bytes()),
```

Comment:
This will panic if the tag is not valid per the spec. Is this verified anywhere? I think it makes sense to just use `Tag` instead of `String` at the source, but otherwise you could use `Tag::new_checked`, which will return an error instead of panicking.

Reply:
Hm. No, probably not. I think I would like to make IR capture Tag actually, but as a separate PR.

Reply:
https://github.com/googlefonts/fontmake-rs/issues/214